integration-cli/docker_utils.go

```diff
@@ -396,6 +396,55 @@ func deleteAllContainers() error {
 	return nil
 }
 
+var protectedImages = map[string]struct{}{}
+
+func init() {
+	out, err := exec.Command(dockerBinary, "images").CombinedOutput()
+	if err != nil {
+		panic(err)
+	}
+	lines := strings.Split(string(out), "\n")[1:]
+	for _, l := range lines {
+		if l == "" {
+			continue
+		}
+		fields := strings.Fields(l)
+		imgTag := fields[0] + ":" + fields[1]
+		protectedImages[imgTag] = struct{}{}
```

Comment:
What about dangling images that exist before the test suite runs?  With `DOCKER_TEST_HOST` you can run the `test-integration-cli` suite against a non-dind daemon, and this is actually exactly how the tests are usually run in Windows.  Do we just need to add a disclaimer that if you run the integration-cli tests against a daemon with images that any dangling images are fair game for culling? :smiling_imp: